src/main/java/processor/CommandProcessor.java

```diff
@@ -0,0 +1,215 @@
+package processor;
+
+import exception.DukeException;
+import exception.EmptyActionException;
+import exception.InvalidActionException;
+import exception.InvalidCommandException;
+import task.DeadlineTask;
+import task.EventTask;
+import task.Task;
+import task.TaskList;
+import task.ToDoTask;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.function.Consumer;
+
+/**
+ * Processes command entered by user.
+ *
+ * <p>The 'CommandProcessor' supports operators, supported include: </p>
+ *
+ * <p> (i) processing the String command and run the operation </p>
+ */
+public class CommandProcessor {
+    private TaskList taskList;
+    private final HashMap<String, Consumer<String>> map = setUpCommandMap();
+
+    CommandProcessor(TaskList taskList) {
+        this.taskList = taskList;
+    }
+
+    private HashMap<String, Consumer<String>> setUpCommandMap() {
+        HashMap<String, Consumer<String>> map = new HashMap<>();
+
+        map.put("list", (command) -> listCommand());
+        map.put("done", (command) -> doneCommand(command));
+        map.put("todo", (command) -> toDoCommand(command));
+        map.put("deadline", (command) -> deadlineCommand(command));
+        map.put("event", (command) -> eventCommand(command));
+        map.put("delete",(command) -> deleteCommand(command));
+        map.put("find",(command) -> findCommand(command));
+        return map;
+    }
+
+    /**
+     * Run command enter by user.
+     * Prints error message if command is invalid.
+     * @param command the command entered by user.
+     */
+    public void runCommand(String command) {
+        Consumer<String> action =  map.get(command.replaceAll(" .*", ""));
+        try {
+            if (action == null) {
+                throw new InvalidCommandException();
+            } else {
+                action.accept(command);
+            }
+        } catch (DukeException e){
```

Comment:
Could perhaps add a more descriptive sentence eg. "An error was thrown: " since the user will be seeing the error

Reply:
Noted. Thanks for the suggestion.